### ISSUE TITLE
cannot access metadata: tags information from gatsby-source-contentful plugin

### ISSUE THREAD

**Issue by MakaBaca** [gatsbyjs/gatsby#36962]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

This is similar to https://github.com/gatsbyjs/gatsby/issues/34556, I cannot access metadata: tags information

**gatsby-config.js**

```js
//Contentful 
          {
                resolve: "gatsby-source-contentful",
                options: {
                  downloadLocal: true,
                  spaceId: process.env.CONTENTFUL_SPACE_ID,
                  accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
                  host: process.env.CONTENTFUL_HOST,
                  enableTags: true,
                },
          }
```

when I run the below query in  graphiQL

```graphql
          query FilterByTagsQuery {
            allContentfulAboutPage(
              sort: { fields: contentful_id }
              filter: {
                metadata: {
                  tags: { elemMatch: { contentful_id: { eq: "test" } } }
                }
              }
            ) {
              nodes {
                title
              }
            }
          }
```

I receive the  below error

```shell
"message": "Field \"metadata\" is not defined by type \"ContentfulAboutPageFilterInput\"."
```

### Reproduction Link

https://github.com/MakaBaca/Contentful

### Steps to Reproduce

1. start he application using yarn start or gatsby develop
2. run the query above in Gatsby GraphQl
3. open (http://localhost:8000/___graphql) in browser and run the query below

```graphql
               query{
                allContentfulAboutPage(
                 filter: {metadata: { tags: { in: ["Test"] } }}) {
                edges { node { title }}}}
```

### Expected Result

The query return back some results if exists else null value.

### Actual Result

The query fails to execute mentioning the error described above.

### Environment

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 5500U with Radeon Graphics         
  Binaries:
    Node: 16.17.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.15.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.674.0), Chromium (107.0.1418.26)
  npmPackages:
    gatsby: ^4.23.1 => 4.24.7 
    gatsby-plugin-image: ^2.23.1 => 2.24.0 
    gatsby-plugin-manifest: ^4.6.0 => 4.24.0 
    gatsby-plugin-react-helmet: ^5.5.0 => 5.24.0 
    gatsby-plugin-sharp: ^4.23.1 => 4.24.0 
    gatsby-plugin-vanilla-extract: ^2.0.1 => 2.0.1 
    gatsby-provision-contentful: ^0.0.3 => 0.0.3 
    gatsby-source-contentful: ^7.21.1 => 7.22.0 
    gatsby-source-filesystem: ^4.22.0 => 4.24.0 
    gatsby-transformer-sharp: ^4.23.1 => 4.24.0 
  npmGlobalPackages:
    gatsby-cli: 4.24.0
```

### Config Flags

```js
// support for .env, .env.development, and .env.production
require("dotenv").config()
require("dotenv").config({
  path: `.env.${process.env.NODE_ENV}`,
})

module.exports = {
  siteMetadata: {
    siteUrl: "https://gatsbycontentfulhomepage.gatsbyjs.io/",
    title: "Gatsby Contentful Homepage Starter",
    author: `Gatsby`,
    description: "A Gatsby Starter for building homepages with Contentful",
  },
  plugins: [
    {
      resolve: "gatsby-source-contentful",
      options: {
        downloadLocal: true,
        spaceId: process.env.CONTENTFUL_SPACE_ID,
        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
        host: process.env.CONTENTFUL_HOST,
        enableTags: true,
      },
    },
    "gatsby-plugin-sharp",
    "gatsby-plugin-image",
    "gatsby-transformer-sharp",
    "gatsby-plugin-react-helmet",
    "gatsby-plugin-vanilla-extract",
    {
      resolve: "gatsby-plugin-manifest",
      options: {
        name: "Gatsby Starter Contentful Homepage",
        short_name: "Gatsby",
        start_url: "/",
        // These can be imported once ESM support lands
        background_color: "#ffe491",
        theme_color: "#004ca3",
        icon: "src/favicon.png",
      },
    },
  ],
}
```

**Comment by matttturnbull** [comment]
Same here, any idea if there's a workaround for this in the mean time?

**Comment by GregPetropoulos** [comment]
This was the work around I used: 
- First check if your tags are being published in Contentful
- Second check your source tab to make sure your getting back strings see screenshot
<img width="1253" alt="Screen Shot 2023-01-15 at 6 47 59 PM" src="https://user-images.githubusercontent.com/68525155/212576696-7c9f5055-fad0-4c23-ac86-fadc85d2930c.png">

- Lastly loop through the  tag strings, in my case I had two loop one for title, subtitle, etc and the second loop for the tags 

```javascript 
     ...<div className='news-list__story-content'>
               <div className='news-list__story-date'>{item.date}</div>
              <h2>Tags: {[item.tags].map(taggedItem=> taggedItem)}</h2>
                     {item.subtitle && <h3>{item.subtitle}</h3>}
      </div>...
```

**Comment by matttturnbull** [comment]
Appreciate it! But I think like @MakaBaca is experiencing, I can't write a query via Gatsby that includes the `tags` field, the response is 'Cannot query field "tags" on type "Contentful..."'. I've double checked that my tags are published and public via the Contentful dash. I'm also able to access them via a direct query to the Contentful GraphQL api, for example:

POST to https://graphql.contentful.com/content/v1/spaces/XXX

`query getPosts {
  postCollection {
    items {
      title
      sys {
        id
      }
      contentfulMetadata {
        tags {
          id
          name
        }
      }
    }
  }
}`

**Comment by GregPetropoulos** [comment]
My package.json We have different versions this might be the issue, also I am wondering if in your dashboard you added a text field in your content-types and named it tags? if so that could be the culprit.
```json
{
    "scripts": {
    "develop": "gatsby develop",
    "start": "gatsby develop",
    "build": "gatsby build",
    "serve": "gatsby serve",
    "clean": "gatsby clean",
    "typecheck": "tsc --noEmit",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "@emotion/react": "^11.10.5",
    "@mantine/core": "^5.9.4",
    "@mantine/hooks": "^5.9.4",
    "@types/localizejs-library": "^1.0.3",
    "classnames": "^2.3.2",
    "dayjs": "^1.11.7",
    "gatsby": "^5.2.0",
    "gatsby-omni-font-loader": "^2.0.2",
    "gatsby-plugin-google-analytics": "^5.2.0",
    "gatsby-plugin-image": "^3.2.0",
    "gatsby-plugin-manifest": "^5.3.1",
    "gatsby-plugin-react-helmet": "^6.3.0",
    "gatsby-plugin-sass": "^6.2.0",
    "gatsby-plugin-sharp": "^5.2.0",
    "gatsby-source-contentful": "^8.4.0",
    "gatsby-source-filesystem": "^5.2.0",
    "gatsby-transformer-sharp": "^5.2.0",
    "leaflet": "^1.9.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-helmet": "^6.1.0",
    "react-hook-form": "^7.41.1",
    "react-leaflet": "^4.2.0",
    "react-phone-input-2": "^2.15.1",
    "sass": "^1.56.1"
  },
  "devDependencies": {
    "@babel/core": "^7.20.5",
    "@storybook/addon-actions": "^6.5.14",
    "@storybook/addon-essentials": "^6.5.14",
    "@storybook/addon-interactions": "^6.5.14",
    "@storybook/addon-links": "^6.5.14",
    "@storybook/builder-webpack5": "^6.5.14",
    "@storybook/manager-webpack5": "^6.5.14",
    "@storybook/react": "^6.5.14",
    "@storybook/testing-library": "^0.0.13",
    "@types/leaflet": "^1.9.0",
    "@types/node": "^18.11.10",
    "@types/react": "^18.0.26",
    "@types/react-dom": "^18.0.9",
    "@types/react-helmet": "^6.1.6",
    "babel-loader": "^8.3.0",
    "prettier": "2.8.1",
    "typescript": "^4.9.3"
  }
}
```
My terminal if that helps...
<img width="438" alt="Screen Shot 2023-01-15 at 10 34 35 PM" src="https://user-images.githubusercontent.com/68525155/212593211-ce0a1698-c74b-483f-8130-27316753b145.png">

**Comment by matttturnbull** [comment]
Hmmm nice, I'll try with those versions tomorrow, currently I'm:

```
  "dependencies": {
    ...
    "gatsby": "^5.4.1",
    "gatsby-source-contentful": "^8.4.1",
    ...
  }
```
Terminal:
<img width="306" alt="Screenshot 2023-01-16 at 6 15 15 PM" src="https://user-images.githubusercontent.com/33284906/212603602-17f40819-2b25-4f08-92a1-295e900e325f.png">

In Contentful, at one point I may have tried my own text field named `tags`, however I removed that when I discovered the panel they've created 🤦‍♀️:

<img width="835" alt="Screenshot 2023-01-16 at 6 10 52 PM" src="https://user-images.githubusercontent.com/33284906/212603350-f867a11b-1f9c-49b7-b216-c5e5fa915161.png">

**Comment by matttturnbull** [comment]
Cracked it, down to a silly error... I had `enableTags: true` outside of the `options` object in gatsby-config.js. Doh!

**Comment by LekoArts** [comment]
I can't use the reproduction linked in the original post (no credentials). So was your issue solved by publishing the tags? @matttturnbull @GregPetropoulos

**Comment by GregPetropoulos** [comment]
@LekoArts My issue was solved by publishing the tags and creating a 2nd loop through on the tags. To be clear these are tags that are made as short string field in the Content-Type, these are not the same ***Tags*** that are created by the Contentful dashboard tabs.
<img width="946" alt="Screen Shot 2023-01-17 at 10 43 01 AM" src="https://user-images.githubusercontent.com/68525155/212943722-9a4bddf9-a6b4-4995-8023-ada785bddbb1.png">

**Comment by LekoArts** [comment]
As this issue seems solved then (+ we can't access the reproduction), I'm closing this.

**Comment by zsolt-halo** [comment]
Anyone having issues with this, I've spent a day or two trying to figure out what's up and I was missing 2 things:

1.: add the required flag to the options object in gatsby config:
```js
options: {
        enableTags: true,
        downloadLocal: true,
        spaceId: process.env.CONTENTFUL_SPACE_ID,
        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
        host: process.env.CONTENTFUL_HOST,
      },
```

2.: Make sure you have published your tags, just addig them to the content types is not enough, press "Republish" on the Contentful admin panel.
